src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,214 @@
+package duke.parser;
+
+import duke.exceptions.InvalidTypeException;
+import duke.exceptions.MissingNameException;
+import duke.exceptions.MissingEventDateException;
+import duke.exceptions.MissingDeadlineDateException;
+import duke.storage.Storage;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+import duke.ui.Ui;
+import java.time.format.DateTimeParseException;
+
+public class Parser {
+
+    private String userInput;
+
+    public Parser(String userInput) {
+        this.userInput = userInput;
+    }
+
+    /**
+     * Process user commands.
+     *
+     * @return False if the command is "bye", True otherwise
+     */
+    public boolean processInput() {
+        if(userInput.equals("bye")) {
```

Comment:
There should be a space between `if` and the open bracket. I noticed the same issue on line 81 and in some of the other files. Perhaps you could consider using [checkStyle](https://nus-cs2103-ay2122s2.github.io/website/schedule/week4/project.html#4-add-increments-as-branches-a-gradle-a-checkstyle-level-10-a-varargs)? It really helps me a lot in fixing my code to meet the coding standards 😅